tests/zfs-tests/tests/functional/tmpfile/tmpfile_001_pos.c

```diff
@@ -0,0 +1,65 @@
+#include <stdio.h>
+#include <stdlib.h>
+#include <unistd.h>
+#include <sys/types.h>
+#include <sys/stat.h>
+#include <fcntl.h>
+#include <errno.h>
+#include <string.h>
+
+#ifndef __O_TMPFILE
+#define	__O_TMPFILE	020000000
+#endif
+
+#ifndef O_TMPFILE
+#define	O_TMPFILE	(__O_TMPFILE|O_DIRECTORY)
+#endif
+
+int
+main(int argc, char *argv[])
+{
+	int i, fd;
+	ssize_t ret;
+	char buf1[64], buf2[64];
+	char *penv[] = {"TESTDIR"};
+
+	(void) fprintf(stdout, "Verify O_TMPFILE is working properly.\n");
+
+	/*
+	 * Get the environment variable values.
+	 */
+	for (i = 0; i < sizeof (penv) / sizeof (char *); i++) {
+		if ((penv[i] = getenv(penv[i])) == NULL) {
```

Comment:
I copy this from `ctime/ctime_001_pos`. The problem is that this is run directly by test runner rather than custom script, so I don't know if we can supply command line option.